### PR TITLE
[DI] add docs for new namespace option

### PR DESCRIPTION
**Pull request by kbond**
| Q             | A
| ------------- | ---
| Doc fix?       | no
| New docs?      | yes
| Applies to?  | 3.4
| Fixed tickets?    | https://github.com/symfony/symfony/pull/23991

### CODE REVIEW

**Review by HeahDude — APPROVED** [review]

File: service_container.rst [hunk 1]
```diff
@@ -923,6 +923,34 @@ them will not cause the container to be rebuilt.
...
+The namespace Option
+--------------------
+
+.. versionadded:: 3.4
+    The ``namespace`` option was added in Symfony 3.4.
+
+When using the ``resource`` option in YAML, the namespace prefix can only be used once
+per file when defining it as the ``id``. In order to have multiple definitions in the
+same file with the same namespace prefix, you can use the ``namespace`` option. When
+this option is used, the ``id`` can be anything as long as it is unique. For example,
```
Comment (xabbuh) [inline service_container.rst]: IMO this sentence is confusing. One may ask: Where is the id now taken from?

File: service_container.rst [hunk 2]
```diff
@@ -923,6 +923,34 @@ them will not cause the container to be rebuilt.
...
+you can define your services like this:
+
+.. configuration-block::
```
Comment (xabbuh) [inline service_container.rst]: We can omit the configuration block and just start with the YAML code block.

File: service_container.rst [hunk 3]
```diff
@@ -923,6 +923,34 @@ them will not cause the container to be rebuilt.
     means that all classes are "available to be *used* as services" without needing
     to be manually configured.
 
+The namespace Option
```
Comment (xabbuh) [inline service_container.rst]: What about enclosing the following section in a `sidebar` directive instead? IMO it's a nice addition for the YAML format, but not a totally new concept.

File: service_container.rst (hunk 2 above)
Comment (kbond) [reply service_container.rst]: Done.

File: service_container.rst (hunk 3 above)
Comment (kbond) [reply service_container.rst]: Done.

File: service_container.rst (hunk 1 above)
Comment (kbond) [reply service_container.rst]: Should I remove the sentence?

**Comment by weaverryan** [comment]
@kbond Can you tell me when this is a useful feature people (I didn't meant that to sound negative at all - I want to make sure I understand the precise use-case). Is it only when there are multiple classes in the same file? If not, I didn't quite understand the description of it.

I would also prefer to put this doc somewhere else (though I'm not sure where). I want to keep the main article very light and focused on the 90% use-cases.

**Comment by kbond** [comment]
Here is a use-case: https://github.com/SimpleBus/SymfonyBridge/pull/43#issuecomment-325676233

**Comment by kbond** [comment]
FYI - XML has the namespace option already. I can find reference to it on this page: http://symfony.com/doc/current/service_container/3.3-di-changes.html. Don't know if it makes sense to move there as this is a 3.4 change.

**Comment by javiereguiluz** [comment]
@kbond thanks for this contribution! I'm sorry we didn't merge it earlier. The doc your proposed was great ... but we try to not use sidebars in the docs (because you can't easily link to them directly) so I reworded the contents to be a normal section instead. Thanks!

**Pull request closed without merging**